### ISSUE TITLE
Windows 10, boot dev  -- process cannot access file in temp.

### ISSUE THREAD

**Issue by tensor-programming** [boot-clj/boot#649]
Running emacs on windows 10.  Decided to install boot and try out Hoplon.  Basically, when I run `boot dev` I get a few specific errors when changing the code inside of the emacs.  The main errors are `The process cannot access the file because it is being used by another process.` and ` java.io.IOException: Couldn't delete C:\Users\user-name\.boot\cache\tmp\Projects\clojure\project-name\421\d82h8\file.html`  which hangs `boot dev` until i ctr-c and then rerun the command. 

Oddly enough the errors start as soon as I type anything into emacs.  Even just hitting return or adding a space will trigger this error if boot dev is running.  Ill get an error like `java.lang.UnsupportedOperationException: nth not supported on this type: Symbol` then it will come back and say `writing HTML files... index.html` then the thread will hang and give me the `used by another process.` & `cannot delete file` errors again.  

I get these errors with completely valid clojure code that will compile with boot if I rerun `boot dev`. 

I've tried using an elevated cmd prompt as well as a different editor but I still get the same errors in Visual Studio Code and Atom (and I would really like to be able to use emacs).

**Comment by matthewlal** [comment]
These approaches should help:

If you are using boot-http refer to this page https://github.com/pandeiro/boot-http/wiki/Using-with-windows

If you are using Cursive with IntelliJ navigate to Settings > Appearance & Behaviour > System Settings > Use "safe write" (save changes to a temporary file first)
You want to uncheck that option.

If you have real-time anti-virus scanning disabling it may help.

I also recommend running boot-clj in the Linux Subsystem for Windows (WSL). If you do this repeat step 2 and perhaps step 1 as well if it still doesn't work.

I was running boot-clj in WSL initially. My IFDE (Immediate Feedback Dev Environment) would stop running with exceptions if syntax wasn't correct, this wasn't ideal for me so I started troubleshooting. I installed boot-clj with choco in Windows 10, I was getting errors similar to what I was having before and the issues in this thread. After doing step 2 I had a feeling it would solve my issue, so I tested my IFDE in WSL and everything worked. I prefer working in WSL, so I uninstalled boot-clj in my windows environment.
